opentelemetry-sdk/src/metrics/data/mod.rs

```diff
@@ -18,7 +18,7 @@ pub struct ResourceMetrics {
 }
 
 /// A collection of metrics produced by a meter.
-#[derive(Default, Debug)]
+#[derive(Default, Clone, Debug)]
```

Comment:
what is the need to do Clone? Is it just for InMemoryExporter's need ?

Reply:
For the most part yes :)
Previously we had `Box<dyn Aggregation>` which prevented other types from being cloneable as well.
Since AggregatedMetrics is simple enum, I made other types cloneable as well.
I'm not sure if it is valuable to have `Clone` on these types, but given that end users will use them, maybe someone will find it useful :)

Reply:
I see. Then my suggestion is *not* to implement Clone now (or do it when testing feature is enabled only so only InMemoryExporter can get it). Adding clone in future is always possible, but reversing that decision is not allowed as it'll be breaking change.

(Exporters usually export the metric in the export() call itself, and even if it gets stored for retry etc. it'll be in a serialized state, so I don't think Clone is needed)

Reply:
`Clone` removed.